idaes/models_extra/power_generation/unit_models/soc_submodels/electrode.py

```diff
@@ -0,0 +1,952 @@
+                for i in comps:
+                    _set_if_unfixed(
+                        self.conc_mol_comp_deviation_x[t, self.ixnodes.first(), iz, i],
+                        self.conc_mol_comp_deviation_x0[t, iz, i],
+                    )
+                for ix in self.ixnodes:
+                    if temperature_guess is not None:
+                        _set_if_unfixed(self.temperature_z[t, iz], temperature_guess)
+                        _set_if_unfixed(self.temperature_deviation_x[t, ix, iz], 0)
+                        _set_if_unfixed(self.temperature_deviation_x0[t, iz], 0)
+                        _set_if_unfixed(self.temperature_deviation_x1[t, iz], 0)
+                    if pressure_guess is not None:
+                        _set_if_unfixed(self.pressure[t, ix, iz], pressure_guess)
+                    for i in comps:
+                        _set_if_unfixed(self.conc_mol_comp_deviation_x[t, ix, iz, i], 0)
+                    mol_dens = pyo.value(
+                        sum(self.conc_mol_comp[t, ix, iz, i] for i in comps)
+                    )
+                    _set_if_unfixed(
+                        self.pressure[t, ix, iz],
+                        _constR * self.temperature[t, ix, iz] * mol_dens,
+                    )
+                    for i in comps:
+                        _set_if_unfixed(
+                            self.mole_frac_comp[t, ix, iz, i],
+                            self.conc_mol_comp[t, ix, iz, i] / mol_dens,
+                        )
+                    _set_if_unfixed(
+                        self.enth_mol[t, ix, iz],
+                        sum(
+                            common._comp_enthalpy_expr(self.temperature[t, ix, iz], i)
+                            * self.mole_frac_comp[t, ix, iz, i]
+                            for i in comps
+                        ),
+                    )
+                    if self.config.has_holdup:
+                        _set_if_unfixed(
+                            self.int_energy_mol[t, ix, iz],
+                            sum(
+                                common._comp_int_energy_expr(
+                                    self.temperature[t, ix, iz], i
+                                )
+                                * self.mole_frac_comp[t, ix, iz, i]
+                                for i in comps
+                            ),
+                        )
+                        _set_if_unfixed(
+                            self.int_energy_density[t, ix, iz],
+                            self.int_energy_mol[t, ix, iz] / self.vol_mol[t, ix, iz],
+                        )
+                for i in comps:
+                    _set_if_unfixed(
+                        self.conc_mol_comp_deviation_x1[t, iz, i],
+                        self.conc_mol_comp_deviation_x[t, self.ixnodes.last(), iz, i],
+                    )
+
+        opt = get_solver(solver, optarg)
+        common._init_solve_block(self, opt, solve_log)
```

Comment:
I am not sure if `_init_solve_block` does this already, but this should raise an `InitializationError` if if fails to solve. I would suggest that it should return a solver status that you can check for optimal termination and then raise an exception if required.

Reply:
I created this helper function in part because I kept writing these tiny blocks of code raising initialization errors 1) if the problem was nonsquare somehow or 2) if the solve failed.  Now it does both of those things (and redirects the output to `solve_log`.